packages/core/test/lint-core.test.ts

```diff
@@ -388,13 +388,11 @@ describe('rules', () => {
       rawSchema: given.tokens,
       rules: [{ id: given.rule, severity: 'error', options: given.options }],
     });
-    if (notices?.length) {
```

Comment:
This is actually not what we want—this relies too much on output and skips over the case where we _wanted_ an error but none was thrown.

Instead, prioritize the `want` cases first, and then assert the results match